### PR TITLE
Make TCK build cache location configurable

### LINKED ISSUE #25392: TCK use of mvn wrapper failing in new CI system
#25191 updated `TCKRunner` to use the mvn wrapper to launch TCK tests. When running in the CI build, it sets up a temporary maven home directory where it stores a custom settings.xml and the local repository.

The current approach is to walk up the directory hierarchy looking for the `dev` workspace directory, but this isn't present in the new CI build.

We should instead read a new property (`fat.test.build.cache.dir`) which will be set for us in the new CI system.

### PR DESCRIPTION
**Pull request by Azquelt**
Allow the build system to supply a location to use for caching artifacts for the life of the build. This is used for storing the temporary maven home directory for TCK tests.

Fixes #25392

### CODE REVIEW

**Comment by Azquelt** [comment]
#build
#pb

File: dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java [hunk 1]
```diff
@@ -708,32 +709,58 @@ public static String getArtifactoryToken() {
...
+     * In the development environment, this will be the workspace {@code dev} directory
+     *
+     * @return             the build cache directory
+     * @throws IOException
+     */
+    public static File getBuildCacheDir() throws IOException {
+        String buildCacheDirPath = System.getProperty(BUILD_CACHE_DIR_PROPERTY);
+
+        File cacheDir;
+        if (!buildCacheDirPath.isEmpty()) {
```
Comment (websterc87) [inline dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java]: Should also have a null check incase the property is not set

**Review by websterc87 — CHANGES_REQUESTED** [review]

File: dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java (hunk 1 above)
Comment (tevans78) [inline dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java]: null check needed

**Review by tevans78 — DISMISSED** [review]

**Comment by Azquelt** [comment]
#build
#pb

**Comment by Azquelt** [comment]
#libby

**Comment by Azquelt** [comment]
Rebased on #25422 to pick up that configuration change.

**Comment by Azquelt** [comment]
#pb

**Comment by Azquelt** [comment]
#libby

**Comment by Azquelt** [comment]
Failures in the build are known issues not related to this change.

**Comment by websterc87** [comment]
#pbbeta Requesting one more build so we can see the changes running the children in jenkins.

**Review by websterc87 — COMMENTED** [review]
Changes look good but would like to run it via the Jenkins Children to make sure they dont break again. I should have just requested it is github events are working.

**Review by websterc87 — DISMISSED** [review]
Changes look good and pass happily run in jenkins children.

This issues in the latest build are unrelated to this PR and related to a Java17 change and defect: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=296309

A fix for this was delivered on the 15th but wouldnt have been pulled into this PR.

**Review by websterc87 — DISMISSED** [review]
Happy to merge

**Comment by Azquelt** [comment]
Not sure why github is immediately knocking off the review. It may be because the first commit was already merged in #25422?

Rebased on integration to see if that fixes it.

**Review by websterc87 — APPROVED** [review]

**Comment by Azquelt** [comment]
#libby

**Pull request merged**